feat: form

lvancraen opened the pull request:
### Why

You've reached the form 🎉 

This is PR 8 of 10 that will constitute a base of the forms components and its first use-case -- Patient Update Form.

This PR adds the form. It's based on the generic form in Kariko with slight alterations. I've removed certain features that aren't used yet so as to not clutter the code. I've also added a way of providing custom validation to a schema via Zod

### Ticket

None (technically there is one for the patient update form but I'll only assign it to the last PR)

### Accomplishments

- [x] Adds Form

LUMacKenn reviewed (approved):
I don't have a ton of context on this, I assume it's mostly just copy/paste from kariko, maybe @danstans want to take a look?

lvancraen commented:
> I don't have a ton of context on this, I assume it's mostly just copy/paste from kariko, maybe @danstans want to take a look?

It is mostly copy paste. I trimmed it down a little by removing features we don't use. I refactored the form structuring code. And I added conditional and custom field validation.
